snowpack/src/scan-imports.ts

```diff
@@ -155,23 +152,38 @@ function parseFileForInstallTargets({
   locOnDisk,
   baseExt,
   contents,
-}: SnowpackSourceFile): InstallTarget[] {
+}: SnowpackSourceFile<string>): InstallTarget[] {
+  const relativeLoc = path.relative(process.cwd(), locOnDisk);
+
   try {
     switch (baseExt) {
       case '.css':
       case '.less':
       case '.sass':
       case '.scss': {
+        logger.debug(`Scanning ${relativeLoc} for imports as CSS`);
         return parseCssForInstallTargets(contents);
       }
       case '.html':
       case '.svelte':
       case '.vue': {
+        logger.debug(`Scanning ${relativeLoc} for imports as HTML`);
         return parseJsForInstallTargets(extractJSFromHTML({contents, baseExt}));
       }
-      default: {
+      case '.js':
+      case '.jsx':
+      case '.mjs':
+      case '.ts':
+      case '.tsx': {
```

Comment:
Now we’re more explicit in when we scan JS files for imports.

Reply:
Not opposed to this as a larger direction, but in the interest of not unintentionally breaking users could we instead use that "is binary data" check so that we don't introduce a huge change here? Considering that this will most likely go out in a bug fix, I'd like to minimize the change if possible.

Reply:
since we're doing the filter below as well, could we walk this switch statement change back out and still fix the breaking issue for 2.10.2, and then re-introduce the switch statement change in 2.11.0? I'd hate to release a patch release for this that then unnecessarily breaks someone else.

Reply:
We actually can’t, because we load string files like `.svg` and `.txt` and are parsing them for imports. Before we were passing more things to `parseJsForInstallTargets()` than I think we realized. So yes, we can filter out binary files, but we also need to be more explicit with JS files too.

Reply:
Also to be clear, the loading of `.svg` and `.txt` files probably shouldn’t happen, but I think that trying to change that would be a riskier change.